### PR TITLE
Fix parquet row group prefetching and erasing

### PR DESCRIPTION
**Pull request by marin-ma**
Summary: This PR addresses two main issues related to row group management:

Details:
1. Row group prefetching issue. The existing condition to check if a row group has been loaded before is incorrect, which prevents the row group from being prefetched as configured.

2. First row group never get erased. Due to a bug, the first buffered row group never gets erased after loading subsequent row groups.

Fixes: #6115

### CODE REVIEW

**Comment by marin-ma** [comment]
@majetideepak I'm wondering if this is also a bug? Seems like `rowGroupIds[0]` will never get erased from `inputs_`

https://github.com/facebookincubator/velox/blob/fcd334c97c2aff6a311702948343e93e80297976/velox/dwio/parquet/reader/ParquetReader.cpp#L594-L596

File: velox/dwio/parquet/tests/reader/ParquetReaderTest.cpp
Comment (majetideepak) [inline velox/dwio/parquet/tests/reader/ParquetReaderTest.cpp:27]: We should make this a parquet reader option.
Flags are discouraged.

**Comment by majetideepak** [comment]
> @majetideepak I'm wondering if this is also a bug? Seems like `rowGroupIds[0]` will never get erased from `inputs_`

@marin-ma You are right. The check should be `if (currentGroup >= 1)`

File: velox/dwio/parquet/tests/reader/ParquetReaderTest.cpp
Comment (majetideepak) [inline velox/dwio/parquet/tests/reader/ParquetReaderTest.cpp:632]: We can probably add `EXPECT_FALSE(parquetRowReader->isRowGroupBuffered(0));` here after fixing the check

**Comment by marin-ma** [comment]
@majetideepak Also updated issue/PR description. PTAL. Thanks!

File: velox/docs/develop/TpchBenchmark.rst
Comment (majetideepak) [inline velox/docs/develop/TpchBenchmark.rst:118]: Let's keep this and add a flag `parquet_prefetch_rowgroups` to TpchBenchmark.cpp.
We can then pass this via the connector properties just like `max_coalesced_bytes`
Also please fix typo here `max_coalesce_bytes` -> `max-coalesced-bytes`
`max_coalesce_distance_bytes` -> `max_coalesced_distance_bytes`

**Review by majetideepak — COMMENTED** [review]
@marin-ma Looks great! Just one comment.

**Review by majetideepak — APPROVED** [review]
@marin-ma thanks! Can you rebase with the main branch and resolve the conflicts?

**Comment by majetideepak** [comment]
@Yuhta can you take a look?

File: velox/dwio/common/Options.h
Comment (jinchengchenghh) [inline velox/dwio/common/Options.h:469]: Please fix the code style as https://github.com/facebookincubator/velox/blob/main/CODING_STYLE.md#comments

Here we should use //

File: velox/dwio/parquet/tests/reader/ParquetReaderTest.cpp
Comment (jinchengchenghh) [inline velox/dwio/parquet/tests/reader/ParquetReaderTest.cpp:640]: Add the dot in the end of comment

**Comment by marin-ma** [comment]
@Yuhta Could you please take a look? Seems like the CI failure is not related to this PR.

**Comment by marin-ma** [comment]
@majetideepak @Yuhta Could you please take a look? Can this one get merged?

**Comment by marin-ma** [comment]
@Yuhta Could you please take a look?

**Comment by marin-ma** [comment]
@mbasmanova Could you take a look?

File: velox/dwio/parquet/reader/ParquetReader.h
Comment (mbasmanova) [inline velox/dwio/parquet/reader/ParquetReader.h:67]: It would be nice to document this API.

File: velox/dwio/parquet/tests/reader/ParquetReaderTest.cpp
Comment (mbasmanova) [inline velox/dwio/parquet/tests/reader/ParquetReaderTest.cpp:642]: Should we test some other values as well? E.g. 2 (less than total), 4 (exactly as total), 10 (more than total)?

**Review by mbasmanova — COMMENTED** [review]
@marin-ma Looks good to me % a comment about testing. It would be nice to update PR description as it is hard to read and understand. Let's wait for @Yuhta to do a final pass and merge.

**Comment by majetideepak** [comment]
@marin-ma Can you share an update on wrapping this up? Thanks.

**Comment by marin-ma** [comment]
> @marin-ma Can you share an update on wrapping this up? Thanks.

@majetideepak Addressed comments and rebased. Still waiting for @Yuhta's final review. 

@Yuhta Could you help to review? Thanks!

**Comment by majetideepak** [comment]
@Yuhta, @mbasmanova any further feedback? Thanks.

File: velox/dwio/parquet/reader/ParquetReader.cpp
Comment (Yohahaha) [inline velox/dwio/parquet/reader/ParquetReader.cpp:579]: could we use `isRowGroupBuffered` to replace this check?

File: velox/dwio/parquet/tests/reader/ParquetReaderTest.cpp
Comment (Yohahaha) [inline velox/dwio/parquet/tests/reader/ParquetReaderTest.cpp:643]: should we also set directorySizeGuess with a small size? such as 1024, 
 kDefaultDirectorySizeGuess is 1MB.

```c++
  bool preloadFile =
      fileLength_ <= std::max(filePreloadThreshold_, directorySizeGuess_);
```

**Comment by majetideepak** [comment]
@Yuhta Can you please review this PR? Thanks.

**Pull request closed without merging**
